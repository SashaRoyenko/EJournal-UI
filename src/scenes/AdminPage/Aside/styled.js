import styled from 'styled-components';
import { Link } from "react-router-dom";
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  overflow: hidden;
  margin-right: 20px;
  border-radius: 5px;
  width: 190px;
  height: calc(100vh - 215px);
  background-color: ${cssConst.themeMedium};
`;

export const MenuItem = styled(Link)`
  display: block;
  border-bottom: solid 1px ${cssConst.themeLight};
  padding: 10px 15px;
  font-size: 16px;
  font-family: Roboto;
  text-decoration: none;
  color: ${cssConst.labelHighlight};

  &:active {
    background: ${cssConst.themeLightMedium};
  }
`;
