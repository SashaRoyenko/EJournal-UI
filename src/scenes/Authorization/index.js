import React from 'react';
import RegistrationForm from '../../components/RegistrationForm/index';
import LoginForm from '../../components/LoginForm/index';
import {
  PageContainer,
} from './styled';

export default class Authorization extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      registration: false,
    };

    this.switchForm = this.switchForm.bind(this);
  }

  switchForm() {
    this.setState((state) => ({ registration: !state.registration }));
  }

  render() {
    const { registration } = this.state;
    return (
      <PageContainer>
        {
            registration ? (
              <RegistrationForm onClickMessageHandler={this.switchForm} />
            ) : (
              <LoginForm onClickMessageHandler={this.switchForm} />
            )
        }
      </PageContainer>
    );
  }
}
