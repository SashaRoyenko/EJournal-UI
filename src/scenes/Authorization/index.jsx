import React from 'react';
import RegistrationForm from '../../components/RegistrationForm/index.jsx';
import LoginForm from '../../components/LoginForm/index.jsx';
import {
    PageContainer,
} from './styled.js';

export default class Authorization extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            registration: false,
        }

        this.switchForm = this.switchForm.bind(this);
    }

    switchForm() {
        this.setState((state) => ({registration: !state.registration}));
    }

    render() {
        return (
            <PageContainer>
                {
                    this.state.registration ? (
                        <RegistrationForm onClickMessageHandler={this.switchForm}/>
                    ) : (
                        <LoginForm onClickMessageHandler={this.switchForm}/>
                    )
                }
            </PageContainer>
        );
    }
}
