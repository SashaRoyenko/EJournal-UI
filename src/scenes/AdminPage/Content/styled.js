import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  width: calc(100% - 210px);
  height: calc(100vh - 215px);
  padding: 12px 15px 20px;
`;

export const Header = styled.p`
  margin-bottom: 10px;
  font-size: 28px;
  font-family: Roboto;
  font-weight: bold;
  color: ${cssConst.themeDark};
`;
