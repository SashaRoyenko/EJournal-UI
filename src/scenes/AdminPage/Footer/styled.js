import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 20px;
  box-shadow: 0 8px 6px 0 ${cssConst.lightShadow};
  border-radius: 5px;
  width: 100%;
  height: 50px;
  padding: 15px;
  background-color: ${cssConst.themeDark};
`;

export const Label = styled.span`
  font-size: 14px;
  font-family: Roboto;
  color: ${cssConst.labelHighlight};
`;