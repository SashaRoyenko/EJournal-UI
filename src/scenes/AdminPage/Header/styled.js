import styled from 'styled-components';
import { Link } from "react-router-dom";
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 15px;
  box-shadow: 0 8px 6px 0 ${cssConst.lightShadow};
  border-radius: 5px;
  width: 100%;
  height: 100px;
  padding: 15px;
  background-color: ${cssConst.themeDark};
`;

export const HomeLink = styled(Link)`
  font-size: 34px;
  font-family: Roboto;
  text-decoration: none;
  color: ${cssConst.labelHighlight};
`;

export const LoginWrapper = styled.div`
  width: 240px;
`;

export const Label = styled.span`
  font-size: 16px;
  font-family: Roboto;
  font-weight: ${(p) => (p.highlighted ? 'bold' : 'normal')};
  text-decoration: ${(p) => (p.highlighted ? 'underline' : 'none')};
  color: ${(p) => (p.highlighted ? cssConst.labelHighlight : cssConst.labelDefault)};
`;

export const LogOutButton = styled.button`
  margin: 4px 0 0;
  border: none;
  padding: 0;
  background: none;
  font-size: 16px;
  font-family: Roboto;
  color: ${cssConst.labelHighlight};

  &:hover{
    text-decoration: underline;
  }
`;
