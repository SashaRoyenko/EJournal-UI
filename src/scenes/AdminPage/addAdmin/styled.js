import styled from 'styled-components';
import * as cssConst from '../../../shared/variables';

export const Wrapper = styled.div`
  width: calc(100% - 210px);
  height: calc(100vh - 215px);
  padding: 12px 15px 20px 0px;
  border-radius: 5px;
`;

export const ItemTitle = styled.p`
  padding-bottom:10px;
`;

export const TableItem = styled.td`
  padding: 10px 30px 34px 0px;
  margin-bottom: 10px;
  font-size: 16px;
  font-family: Roboto;
  font-weight: bold;
  color: ${cssConst.themeDark};
`;

export const Input = styled.input`
  width: 275px;
  height: 25px;
  border-radius: 3px;
  border: solid 1px #495057;
  background-color: #ffffff;
`;

export const Button = styled.button`
  width: 100px;
  height: 25px;
  border-radius: 7px;
  font-family: Roboto;
  background-color: ${cssConst.buttonBlue};
  font-size: 12px;
  color: white;
  font-weight: bold;
  border:none;
  
  &:active{
  background-color: ${cssConst.buttonBlueOnClick};
  border: 2px solid ${cssConst.turquoise};
  }
`;
