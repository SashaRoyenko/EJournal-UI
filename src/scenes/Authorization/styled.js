import React from 'react'
import styled from 'styled-components'

export const PageContainer = styled.div`
    width: 360px;
    padding: 8% 0 0;
    margin: auto;
`;