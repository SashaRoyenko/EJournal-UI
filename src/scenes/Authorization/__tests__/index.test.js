import React from 'react';
import { mount } from 'enzyme';
import { FormMessage } from '../../../shared/components/Form/styled.js';
import Authorization from '../index.jsx';

describe('<Authorization />', () => {
    it('should render registration form', () => {
        const sut = mount(<Authorization />);

        expect(sut).toMatchSnapshot();
    });

    it('should render login form after click on redirection link', () => {
        const sut = mount(<Authorization />);

        sut.find(FormMessage).find('a').simulate('click');
        
        expect(sut).toMatchSnapshot();
    });
});
