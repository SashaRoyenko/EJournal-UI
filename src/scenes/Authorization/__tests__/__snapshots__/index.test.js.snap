// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authorization /> should render login form after click on redirection link 1`] = `
<Authorization>
  <styled.div>
    <div
      className="sc-AxgMl hhGqQi"
    >
      <Registration
        onClickMessageHandler={[Function]}
      >
        <styled.div>
          <div
            className="sc-AxjAm kFqgcy"
          >
            <styled.img
              alt="logo"
              src={Object {}}
            >
              <img
                alt="logo"
                className="sc-AxhUy fdjSkp"
                src={Object {}}
              />
            </styled.img>
            <form>
              <styled.input
                placeholder="name"
                type="text"
              >
                <input
                  className="sc-AxirZ PTCQu"
                  placeholder="name"
                  type="text"
                />
              </styled.input>
              <styled.input
                placeholder="username"
                type="text"
              >
                <input
                  className="sc-AxirZ PTCQu"
                  placeholder="username"
                  type="text"
                />
              </styled.input>
              <styled.input
                placeholder="password"
                type="password"
              >
                <input
                  className="sc-AxirZ PTCQu"
                  placeholder="password"
                  type="password"
                />
              </styled.input>
              <styled.button>
                <button
                  className="sc-AxiKw bAyasi"
                >
                  create
                </button>
              </styled.button>
              <styled.p>
                <p
                  className="sc-AxhCb eVMfSZ"
                >
                  <span>
                    Not registered? 
                  </span>
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Create an account
                  </a>
                </p>
              </styled.p>
            </form>
          </div>
        </styled.div>
      </Registration>
    </div>
  </styled.div>
</Authorization>
`;

exports[`<Authorization /> should render registration form 1`] = `
<Authorization>
  <styled.div>
    <div
      className="sc-AxgMl hhGqQi"
    >
      <LoginForm
        onClickMessageHandler={[Function]}
      >
        <styled.div>
          <div
            className="sc-AxjAm kFqgcy"
          >
            <styled.img
              alt="logo"
              src={Object {}}
            >
              <img
                alt="logo"
                className="sc-AxhUy fdjSkp"
                src={Object {}}
              />
            </styled.img>
            <form>
              <styled.input
                placeholder="username"
                type="text"
              >
                <input
                  className="sc-AxirZ PTCQu"
                  placeholder="username"
                  type="text"
                />
              </styled.input>
              <styled.input
                placeholder="password"
                type="password"
              >
                <input
                  className="sc-AxirZ PTCQu"
                  placeholder="password"
                  type="password"
                />
              </styled.input>
              <styled.button>
                <button
                  className="sc-AxiKw bAyasi"
                >
                  login
                </button>
              </styled.button>
              <styled.p>
                <p
                  className="sc-AxhCb eVMfSZ"
                >
                  <span>
                    Already registered? 
                  </span>
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Sign In
                  </a>
                </p>
              </styled.p>
            </form>
          </div>
        </styled.div>
      </LoginForm>
    </div>
  </styled.div>
</Authorization>
`;
